src/test/java/sg/dex/starfish/integration/developerTC/AssetRegistration_08.java

```diff
@@ -89,89 +78,4 @@ public void testRegisterWithProv() {
         assertNotNull(asset.getMetadata().get("provenance"));
     }
 
-    @Test
-    public void testHashForResourceAsset() throws IOException {
-
-        // read metadata
-        String asset_metaData = new String(Files.readAllBytes(Paths.get("src/test/resources/assets/SJR8961K_metadata.json")));
-
-        // create asset using metadata and given content
-        ResourceAsset resourceAsset = ResourceAsset.create(asset_metaData, "assets/SJR8961K_content.json");
-        String content = Utils.stringFromStream(resourceAsset.getContentStream());
-        String expected = Hex.toString(Hash.keccak256(content));
-        String actual = resourceAsset.getMetadata().get(Constant.CONTENT_HASH).toString();
-
-        assertEquals(expected, actual);
-    }
-    @Test
-    public void testHashForResourceAssetWithoutMetadata() throws IOException {
-
-        // create asset using metadata and given content
-        ResourceAsset resourceAsset = ResourceAsset.create( "assets/SJR8961K_content.json");
-        String content = Utils.stringFromStream(resourceAsset.getContentStream());
-        String expected = Hex.toString(Hash.keccak256(content));
-        String actual = resourceAsset.getMetadata().get(Constant.CONTENT_HASH).toString();
-
-        assertEquals(expected, actual);
-    }
-
-    @Test(expected = StarfishValidationException.class)
-    public void testBadResourceFile() throws IOException {
-
-        // create asset using metadata and given content
-        ResourceAsset resourceAsset = ResourceAsset.create( "assets/SJR8961K_content_NA.json");
```

Comment:
Why are we deleting this test? We need negative tests for ad assets etc.

I think an interface like `boolean Asset.validateContentHash()` is needed to allow checks to validate any asset type. Default implementation should just use getContentStream and compute the hash as an integrity check.